lib/utils/report.mjs

```diff
@@ -146,3 +134,83 @@ function printfLike(format, ...args) {
 		return result.replace(/%[ds]/, String(arg));
 	}, format);
 }
+
+/** @typedef {import('stylelint').DisabledRangeObject} DisabledRangeObject */
+/** @typedef {import('stylelint').StylelintPostcssResult} StylelintPostcssResult */
+/** @typedef {import('stylelint').Range} Range */
+/** @typedef {import('stylelint').Problem} Problem */
+
+/**
+ * @param {string} ruleName
+ * @param {number} startLine
+ * @param {DisabledRangeObject} [disabledRanges]
+ */
+function isDisabled(ruleName, startLine, disabledRanges) {
+	if (!disabledRanges) return false;
+
+	const ranges = disabledRanges[ruleName] ?? disabledRanges[RULE_NAME_ALL] ?? [];
+
+	for (const range of ranges) {
+		if (
+			// If the problem is within a disabledRange,
+			// and that disabledRange's rules include this one
+			range.start <= startLine &&
+			(range.end === undefined || range.end >= startLine) &&
+			(!range.rules || range.rules.includes(ruleName))
+		) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+/** @param {Problem} problem */
+function isFixApplied(problem) {
+	const { end, start, index, endIndex, fix, node, result, ruleName } = problem;
+	const hasRange = end?.column && start;
```

Comment:
[suggest] Small refactorings:

- The order of start and end seems more natural.
- Checking if the variable is a number seems safer. (since `0` is falsy)

```suggestion
	const hasRange = isNumber(start) && isNumber(end?.column);
```

Reply:
> The first is an `object`.

Sorry, I misunderstood. So, I suggest instead:

```js
const hasRange = Boolean(start) && isNumber(end?.column);
```

Reply:
Ah, `Boolean(start)` emits type errors. Okay, how about this?

```js
const hasRange = start && isNumber(end?.column);
```